pkg/lib/authn/identity/ldap/provider.go

```diff
@@ -25,3 +26,22 @@ func (p *Provider) List(userID string) ([]*identity.LDAP, error) {
 func (p *Provider) GetByServerUserID(serverName string, userIDAttribute string, userIDAttributeValue string) (*identity.LDAP, error) {
 	return p.Store.GetByServerUserID(serverName, userIDAttribute, userIDAttributeValue)
 }
+
+func (p *Provider) New(
+	userID string,
+	serverName string,
+	userIDAttribute string,
+	userIDAttributeValue string,
+	claims map[string]interface{},
+	rawEntryJSON map[string]interface{},
+) *identity.LDAP {
+	return &identity.LDAP{
+		ID:                   uuid.New(),
+		UserID:               userID,
+		ServerName:           serverName,
+		UserIDAttribute:      userIDAttribute,
+		UserIDAttributeValue: userIDAttributeValue,
+		Claims:               claims,
+		RawEntryJSON:         rawEntryJSON,
```

Comment:
Make sure we do not save NULL to  these two JSONB column. I think the best place to do this is in this method.

```
if claims == nil {
  claims = make(map[string]interface{})
}
if rawEntryJSON == nil {
  rawEntryJSON = make(map[string]interface{})
}
```